File: backend/src/main/java/tom/projects/monkeylog/dto/measurement/MeasurementFullResponse.java
package tom.projects.monkeylog.dto.measurement;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class MeasurementFullResponse {
    @NotNull
    private UUID id;
    @NotNull
    private String name;
    @NotNull
    private String unit;
    @NotNull
    private List<MeasurementPointResponse> measurementPoints;
}

File: backend/src/main/kotlin/tom/com/workout/logbook/dto/measurement/MeasurementFullResponse.kt
package tom.com.workout.logbook.dto.measurement

import com.fasterxml.jackson.annotation.JsonUnwrapped

data class MeasurementFullResponse(
    @JsonUnwrapped
    val measurement: MeasurementResponse,
    val points: List<tom.com.workout.logbook.dto.measurement.MeasurementPointResponse>
)